File: aoc2016/day01/src/test/java/de/havox_design/aoc2016/day01/Day01Test.java
package de.havox_design.aoc2016.day01;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day01Test {

    @SuppressWarnings("java:S2699")
    @Test
    void testMainClass() {
        MainClass.main(new String[0]);
    }

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, long expected) {
        Assertions.assertEquals(expected, NoTimeForATaxicab.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("day01Part1sample1.txt", 5L),
                Arguments.of("day01Part1sample2.txt", 2L),
                Arguments.of("day01Part1sample3.txt", 12L)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForPart2")
    void testPart2(String fileName, long expected) {
        Assertions.assertEquals(expected, NoTimeForATaxicab.solvePart2(fileName));
    }

    private static Stream<Arguments> getDataForPart2() {
        return Stream.of(
                Arguments.of("day01Part2sample.txt", 4L)
        );
    }
}

File: aoc2022/aoc2022-kotlin/src/test/kotlin/de/havox_design/aoc2022/day01/Day01Test.kt
package de.havox_design.aoc2022.day01

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day01Test {
    @Test
    fun singleEntry() =
        CaloriesCounter("de/havox_design/aoc2022/day01/day01SingleEntry.txt")
            .processFile()
            .shouldBe("42")


    @Test
    fun singleSum() =
        CaloriesCounter("de/havox_design/aoc2022/day01/day01SingleSum.txt")
            .processFile()
            .shouldBe("797")

    @Test
    fun sample() =
        CaloriesCounter("de/havox_design/aoc2022/day01/day01Sample.txt")
            .processFile()
            .shouldBe("24000")

    @Test
    fun sampleTopThree() =
        CaloriesCounter("de/havox_design/aoc2022/day01/day01Sample.txt")
            .processFileTopThree()
            .shouldBe("45000")
}

private fun String.shouldBe(expectation: String) = assertEquals(expectation, this)
